SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -9445,42 +9306,36 @@ void CEulerSolver::TurboAverageProcess(CSolver **solver, CGeometry *geometry, CC
 void CEulerSolver::MixedOut_Average (CConfig *config, su2double val_init_pressure, const su2double *val_Averaged_Flux,
                                      const su2double *val_normal, su2double& pressure_mix, su2double& density_mix) {
 
-  su2double dx, f, df, resdl = 1.0E+05;
-  unsigned short iter = 0, iDim;
-  su2double relax_factor = config->GetMixedout_Coeff(0);
-  su2double toll = config->GetMixedout_Coeff(1);
+  const auto relax_factor = config->GetMixedout_Coeff(0);
+  const auto toll = config->GetMixedout_Coeff(1);
   unsigned short maxiter = SU2_TYPE::Int(config->GetMixedout_Coeff(2));
-  su2double dhdP, dhdrho, enthalpy_mix, velsq, *vel;
-
-  vel = new su2double[nDim];
 
   pressure_mix = val_init_pressure;
```

Comment:
Same as density_mix, maybe I'm missing something here?

Reply:
density_mix is computed based on the value of pressure_mix, which changes throughout the iterative process. Therefore, a local instance of density_mix can be defined within the scope of the iterative loop.

Reply:
Yes, I'm being stupid I see now. Unsure why the CodeQL is piping up. Maybe it wants su2double &density_mix as the function argument.